books/data_transformation.py

```diff
@@ -1,108 +1,128 @@
-
-
-def transform_scrapped_books_data():
-    scrapped_data = pd.read_csv(os.path.join(raw_data_path, 'scrapped_book_data.csv'), delimiter=';', usecols=['Title', 'Author', 'ISBN/ISSN:', 'Wydawnictwo:', 'Rok wydania:'], low_memory=False)
-    new_scraped_data_column_names = {
-        'ISBN/ISSN:': 'book_isbn',
-        'Title': 'book_title',
-        'Author': 'book_author',
-        'Rok wydania:': 'year_published',
-        'Wydawnictwo:': 'publisher',
-    }
-    scrapped_data.rename(columns=new_scraped_data_column_names, inplace=True)
-    scrapped_data['year_published'] = scrapped_data['year_published'].fillna(0).astype(int)
-    scrapped_data.to_csv(os.path.join(load_path, 'scrapped_books_data_transformation_v1.csv'), sep=';', index=False, header=True)
+import logging
+from argparse import ArgumentParser
+from datetime import datetime
+
+
+logging.basicConfig(filename='api.log', level=logging.INFO,
+                    format='%(asctime)s:%(levelname)s:%(message)s')
+
+
+class DataTransformation:
+
+    def __init__(self):
+        self.data_frames = {}
+        logging.info("DataTransformation initialized")
+
+    def transform_books(self, source_path):
+        books_frame = pd.read_csv(os.path.join(source_path, 'books.csv'), delimiter=';',
+                                  usecols=['ISBN', 'Book-Title', 'Book-Author', 'Year-Of-Publication', 'Publisher', 'Image-URL-S'],
+                                  encoding='windows-1250', low_memory=False)
+
+        new_books_column_names = {
+            'ISBN': 'book_isbn',
+            'Book-Title': 'book_title',
+            'Book-Author': 'book_author',
+            'Year-Of-Publication': 'year_published',
+            'Publisher': 'publisher',
+            'Image-URL-S': 'book_image_url'
+        }
+        books_frame.rename(columns=new_books_column_names, inplace=True)
+        self.data_frames['books'] = books_frame
+
+    def transform_users(self, source_path):
+        users_frame = pd.read_csv(os.path.join(source_path, 'users.csv'), usecols=['Location', 'Age'], delimiter=';', encoding='windows-1250', low_memory=False)
+        df_location = users_frame['Location'].str.split(',', expand=True)
+        df_location = df_location.drop(columns=[1, 3, 4, 5, 6, 7, 8])
+
+        users_frame = users_frame.drop(columns=['Location'])
+        users_frame = pd.concat([users_frame, df_location], axis=1)
+        new_users_column_names = {
+            'Age': 'user_age',
+            0: 'user_city',
+            2: 'user_country'
+        }
+
+        users_frame.rename(columns=new_users_column_names, inplace=True)
+        users_frame['user_city'] = users_frame['user_city'].str.title()
+        users_frame['user_country'] = users_frame['user_country'].str.title()
+        users_frame['user_age'] = users_frame['user_age'].fillna(0).astype(int)
+        self.data_frames['users'] = users_frame
```

Comment:
this should not be here at all. Just save the results to the target file here :)